core/domain/learner_group_services.py

```diff
@@ -0,0 +1,647 @@
+            compared with the title of the story if passed in arguments.
+
+    Returns:
+        list(dict). The matching story syllabus item dicts of the given topic.
+    """
+    story_ids = [
+        story.story_id for story in
+        topic.canonical_story_references
+        if (story.story_id not in group_story_ids
+            and story.story_is_published is True)
+    ]
+
+    matching_stories = story_fetchers.get_story_summaries_by_ids(story_ids)
+    stories = story_fetchers.get_stories_by_ids(story_ids)
+
+    matching_story_syllabus_item_dicts: List[
+        story_domain.LearnerGroupSyllabusStorySummaryDict] = []
+
+    for ind, story in enumerate(matching_stories):
+        if keyword is None or story.title.lower().find(keyword) != -1:
+            summary_dict = story.to_dict()
+            syllabus_story_dict: story_domain.LearnerGroupSyllabusStorySummaryDict = { # pylint: disable=line-too-long
+                'id': summary_dict['id'],
+                'title': summary_dict['title'],
+                'description': summary_dict['description'],
+                'language_code': summary_dict['language_code'],
+                'version': summary_dict['version'],
+                'node_titles': summary_dict['node_titles'],
+                'thumbnail_filename': summary_dict['thumbnail_filename'],
+                'thumbnail_bg_color': summary_dict['thumbnail_bg_color'],
+                'url_fragment': summary_dict['url_fragment'],
+                'story_model_created_on':
+                    summary_dict['story_model_created_on'],
+                'story_model_last_updated':
+                    summary_dict['story_model_last_updated'],
+                'story_is_published': True,
+                'completed_node_titles': [],
+                'all_node_dicts': [
+                    node.to_dict() for node in
+                    stories[ind].story_contents.nodes # type: ignore[union-attr]
+                ],
+                'topic_name': topic.name,
+                'topic_url_fragment': topic.url_fragment
+            }
+            matching_story_syllabus_item_dicts.append(syllabus_story_dict)
+
+    return matching_story_syllabus_item_dicts
+
+
+def add_student_to_learner_group(
+    group_id: str,
+    user_id: str,
+    progress_sharing_permission: bool
+) -> None:
+    """Adds the given student to the given learner group.
+
+    Args:
+        group_id: str. The id of the learner group.
+        user_id: str. The id of the student.
+        progress_sharing_permission: bool. The progress sharing permission of
+            the learner group. True if progress sharing is allowed, False
+            otherwise.
+
+    Raises:
+        Exception. Student was not invited to join the learner group.
+    """
+    learner_group_model = learner_group_models.LearnerGroupModel.get(
+        group_id, strict=False)
+
+    # Ruling out the possibility of None for mypy type checking.
+    assert learner_group_model is not None
```

Comment:
Ditto as above, if we don't expect it to be `None` use `strict=True`

Reply:
done